verification/dependencies/net/ip.gobra

```diff
@@ -126,15 +126,16 @@ func (ip IP) IsGlobalUnicast() bool /*{
 }*/
 
 // To4 converts the IPv4 address ip to a 4-byte representation.
-preserves forall i int :: 0 <= i && i < len(ip) ==> acc(&ip[i], definitions.ReadL15)
-ensures res != nil ==> len(res) == IPv4len
-ensures len(ip) == IPv4len ==> forall i int :: 0 <= i && i < len(ip) ==> &ip[i] == &res[i]
-ensures (len(ip) == IPv6len && isZeros(ip[0:10]) && ip[10] == 255 && ip[11] == 255) ==> res != nil
-ensures (len(ip) == IPv6len && res != nil) ==> (forall i int :: 0 <= i && i < IPv4len ==> &ip[12+i] == &res[i])
-ensures len(ip) != IPv4len && len(ip) != IPv6len ==> res == nil
-ensures len(ip) == IPv4len ==> len(res) == IPv4len
+preserves wildcard ==> forall i int :: { &ip[i] } 0 <= i && i < len(ip) ==> acc(&ip[i], _)
+preserves !wildcard ==> forall i int :: { &ip[i] } 0 <= i && i < len(ip) ==> acc(&ip[i], definitions.ReadL20)
+ensures   res != nil ==> len(res) == IPv4len
+ensures   len(ip) == IPv4len ==> ip === res
+ensures   (len(ip) == IPv6len && isZeros(ip[0:10]) && ip[10] == 255 && ip[11] == 255) ==> res != nil
+ensures   (len(ip) == IPv6len && !(isZeros(ip[0:10]) && ip[10] == 255 && ip[11] == 255)) ==> res == nil
```

Comment:
Couldn't these conditions be simplified to "isConvertibleToIPv4" or sth similar?

Reply:
no because here `ip` is just `[]byte` whereas the ghost functions introduced in `scion_spec.gobra` take `net.Addr` interface as input